library/rtsp/src/main/java/com/google/android/exoplayer2/source/rtsp/RtspMediaSource.java

```diff
@@ -202,7 +217,8 @@ public RtspMediaSource createMediaSource(MediaItem mediaItem) {
               ? new TransferRtpDataChannelFactory(timeoutMs)
               : new UdpDataSourceRtpDataChannelFactory(timeoutMs),
           userAgent,
-          debugLoggingEnabled);
+          debugLoggingEnabled,
+          socketFactory);
```

Comment:
As mentioned above, the test of whether the socketFactory is set should be here, to make the code easier to read.

`socketFactory == null ? SocketFactory.getDefault() : socketFactory`